internal/cosesign1/cosesign1util_test.go

```diff
@@ -0,0 +1,153 @@
+//go:build linux
+// +build linux
+
+package cosesign1
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/veraison/go-cose"
+)
+
+func readFileBytes(filename string) []byte {
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		println("Error reading '" + filename + "': " + string(err.Error()))
+	}
+	if len(content) == 0 {
+		println("Warning: empty file '" + filename + "'")
+	}
+	return content
+}
+
+func readFileString(filename string) string {
+	return string(readFileBytes(filename))
+}
+
+var FragmentRego string
+var FragmentCose []byte
+var LeafPrivatePem string
+var LeafCertPEM string
+var LeafPubkeyPEM string
+var CertChainPEM string
+
+func comparePEMs(pk1pem string, pk2pem string) bool {
+	pk1der := pem2der([]byte(pk1pem))
+	pk2der := pem2der([]byte(pk2pem))
+	return bytes.Equal(pk1der, pk2der)
+}
+
+/*
+	Decode a COSE_Sign1 document and check that we get the expected payload, issuer, keys, certs etc.
+*/
+
+func Test_UnpackAndValidateCannedFragment(t *testing.T) {
+	var unpacked UnpackedCoseSign1
+	unpacked, err := UnpackAndValidateCOSE1CertChain(FragmentCose, nil, nil, false)
+
+	if err != nil {
+		t.Errorf("UnpackAndValidateCOSE1CertChain failed: %s", err.Error())
+	}
+
+	var iss = unpacked.Issuer
+	var feed = unpacked.Feed
+	var pubkey = base64PublicKeyToPEM(unpacked.Pubkey)
+	var pubcert = base64CertToPEM(unpacked.Pubcert)
+	var payload = string(unpacked.Payload[:])
+	var cty = unpacked.ContentType
+
+	if !comparePEMs(pubkey, LeafPubkeyPEM) {
+		t.Error("pubkey did not match")
+	}
+	if !comparePEMs(pubcert, LeafCertPEM) {
+		t.Error("pubcert did not match")
+	}
+	if cty != "application/unknown+json" {
+		t.Error("cty did not match")
+	}
+	if payload != FragmentRego {
+		t.Error("payload did not match")
+	}
+	if iss != "TestIssuer" {
+		t.Error("iss did not match")
+	}
+	if feed != "TestFeed" {
+		t.Error("feed did not match")
+	}
+}
+
+func Test_UnpackAndValidateCannedFragmentCorrupted(t *testing.T) {
+	FragmentCose2 := make([]byte, len(FragmentCose))
+
+	for i := range FragmentCose2 {
+		FragmentCose2[i] = FragmentCose[i]
+	}
```

Comment:
to copy an array by value you can do:

```suggestion
    FragmentCose2 = FragmentCose
```

Reply:
Are you sure this does a value copy? I need not to change the source one and I am about to zap bytes in the copy.

Reply:
seems I needed 
	copy(fragCose, fragmentCose)